src/Tests/Units/EventListener/RequestHeadersForwarderListener.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace M6Web\Bundle\RequestHeadersForwarderGuzzleBundle\Tests\Units\EventListener;
+
+use GuzzleHttp;
+use M6Web\Bundle\RequestHeadersForwarderGuzzleBundle\EventListener\RequestHeadersForwarderListener as TestedClass;
+use M6Web\Bundle\RequestHeadersForwarderGuzzleBundle\Tests\ContainerTrait;
+use mageekguy\atoum;
+use Symfony\Component\HttpFoundation\HeaderBag;
+
+/**
+ * RequestHeadersForwarderListener test class.
+ */
+class RequestHeadersForwarderListener extends atoum\test
+{
+    use ContainerTrait;
+
+    /**
+     * Test onKernelRequest method.
+     */
+    public function testOnKernelRequest()
+    {
+        $this
+            ->given(
+                // We have a mocked guzzle client which is configured to return a 200
+                $guzzleClient = new GuzzleHttp\Client(['handler' => GuzzleHttp\HandlerStack::create(
+                    new GuzzleHttp\Handler\MockHandler([new GuzzleHttp\Psr7\Response(200)])
+                )]),
+
+                // We set a kind of "spy" on the guzzle client to be able to retrieve sent requests headers later.
+                $guzzleClientRequests = [],
+                $guzzleClient->getConfig('handler')->push(GuzzleHttp\Middleware::history($guzzleClientRequests)),
+
+                // We create the request listener configured to forward X-OAUTH-SAMPLE header to the guzzle client
+                $eventListener = new TestedClass([
```

Comment:
What about this syntax? => http://docs.atoum.org/en/latest/how_to_write_test_cases.html?highlight=newtestedinstance#newtestedinstance-testedinstance